packages/aws-cdk/test/workflows.test.ts

```diff
@@ -0,0 +1,175 @@
+import { Bootstrapper } from '../lib/api/bootstrap';
+import { Deployments } from '../lib/api/deployments';
+import { CdkToolkit, Tag } from '../lib/cdk-toolkit';
+import { listWorkflow } from '../lib/workflows';
+
+let cloudExecutable: MockCloudExecutable;
+let bootstrapper: jest.Mocked<Bootstrapper>;
+
+beforeEach(() => {
+  jest.resetAllMocks();
+
+  bootstrapper = instanceMockFrom(Bootstrapper);
+  bootstrapper.bootstrapEnvironment.mockResolvedValue({ noOp: false, outputs: {} } as any);
+
+  cloudExecutable = new MockCloudExecutable({
+    stacks: [
+      MockStack.MOCK_STACK_A,
+      MockStack.MOCK_STACK_B,
+    ],
+  });
+});
+
+describe('list', () => {
+  test('list stacks with no dependencies', async () => {
+    // GIVEN
+    const toolkit = new CdkToolkit({
+      cloudExecutable,
+      configuration: cloudExecutable.configuration,
+      sdkProvider: cloudExecutable.sdkProvider,
+      deployments: new FakeCloudFormation({
+        'Test-Stack-A': { Foo: 'Bar' },
+        'Test-Stack-B': { Baz: 'Zinga!' },
+      }),
+    });
+
+    // WHEN
+    const workflow = await listWorkflow( toolkit, { selectors: ['Test-Stack-A', 'Test-Stack-B'] });
+
+    // THEN
+    expect(JSON.parse(workflow)).toEqual([{
+      id: 'Test-Stack-A',
+      name: 'Test-Stack-A',
+      environment: {
+        account: '123456789012',
+        region: 'bermuda-triangle-1',
+        name: 'aws://123456789012/bermuda-triangle-1',
+      },
+      dependencies: [],
+    },
+    {
+      id: 'Test-Stack-B',
+      name: 'Test-Stack-B',
+      environment: {
+        account: '123456789012',
+        region: 'bermuda-triangle-1',
+        name: 'aws://123456789012/bermuda-triangle-1',
+      },
+      dependencies: [],
+    }]);
+  });
+
+  test('list stacks with dependent stacks', async () => {
+    // GIVEN
+    const toolkit = new CdkToolkit({
+      cloudExecutable: new MockCloudExecutable({
+        stacks: [
+          MockStack.MOCK_STACK_A,
+          {
+            stackName: 'Test-Stack-B',
+            template: { Resources: { TemplateName: 'Test-Stack-B' } },
+            env: 'aws://123456789012/bermuda-triangle-1',
+            metadata: {
+              '/Test-Stack-B': [
+                {
+                  type: cxschema.ArtifactMetadataEntryType.STACK_TAGS,
+                  data: [
+                    { key: 'Baz', value: 'Zinga!' },
+                  ],
+                },
+              ],
+            },
+            depends: ['Test-Stack-A'],
+          },
+        ],
+      }),
+      configuration: cloudExecutable.configuration,
+      sdkProvider: cloudExecutable.sdkProvider,
+      deployments: new FakeCloudFormation({
+        'Test-Stack-A': { Foo: 'Bar' },
+        'Test-Stack-B': { Baz: 'Zinga!' },
+      }),
+    });
+
+    // WHEN
+    const workflow = await listWorkflow( toolkit, { selectors: ['Test-Stack-A', 'Test-Stack-B'] });
+
+    // THEN
+    expect(JSON.parse(workflow)).toEqual([{
```

Comment:
This should fail. The output does not have dependencies within them.

Reply:
Maybe `toEqual` is not doing a deep equal. Maybe this helps: https://jestjs.io/docs/expect#tomatchobjectobject